synapse/common.py

```diff
@@ -408,6 +459,20 @@ def iterfd(fd, size=10000000):
         yield byts
         byts = fd.read(size)
 
+def spin(genr):
+    '''
+    Crank through a generator but discard the yielded values.
+
+    Args:
+        genr: Any generator or iterable valu. This generator is exhausted
+        via a for loop which discards the values.
+
+    Returns:
+        None
+    '''
+    for item in genr:
```

Comment:
We should replace this with ``collections.deque(gen, maxlen=0)`` 
https://gist.github.com/vEpiphyte/e7f7555c229f31769cfe60a2a32706b4
thanks @jnwatson